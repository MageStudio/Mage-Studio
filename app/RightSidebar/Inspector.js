import React from 'react';
import EmptyInspector from './elements/EmptyInspector';
import MeshInspector from './elements/MeshInspector';

class Inspector extends React.Component {

    constructor(props) {
        super(props);
    }

    getContent() {
        const { empty, element = {} } = this.props;

        if (element.isMesh && element.isMesh()) {
            return (
                <MeshInspector
                    position={element.position()}
                    rotation={element.rotation()}
                    scale={element.scale()}
                />
            );
        }

        return <EmptyInspector />;
    }

    render() {
        return (
            <div className='scene-setting'>
                { this.getContent() }
            </div>
        );
    }
}

export default Inspector;
