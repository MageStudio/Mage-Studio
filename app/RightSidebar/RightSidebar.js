import React from 'react';
import {
    connect
} from 'react-redux';
import { Col } from 'antd';

const RightSidebar = ({empty, x, y, z}) => (
    <Col
        span={4}
        className='sidebar'>
        { empty ? 'No Content' :
            <ul>
                <li>{x}</li>
                <li>{y}</li>
                <li>{z}</li>
            </ul> }
    </Col>
);

const mapStateToProps = (state) => {
    const { rightsidebar } = state;
    const { empty, element } = rightsidebar;

    console.log(element);
    const position = (element && element.position()) || { x: 0, y: 0, z: 0 };

    return {
        empty,
        ...position
    }
}

export default connect(mapStateToProps)(RightSidebar);
