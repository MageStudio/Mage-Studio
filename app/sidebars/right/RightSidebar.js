import React from 'react';
import {
    connect
} from 'react-redux';
import { Col, Icon } from 'antd';

import CopyButton from '../../common/CopyButton';
import DeleteButton from '../../common/DeleteButton';
import AddButton from '../../common/AddButton';
import SearchButton from '../../common/SearchButton';

import Hierarchy from './Hierarchy';
import Inspector from './Inspector';

import {
    meshChanged,
    textureChanged
} from '../../actions/currentMesh';

import {
    getScripts,
    getSingleScript
} from '../../actions/scripts';

class RightSidebar extends React.Component {

    constructor(props) {
        super(props);
    }

    handleScriptMount = () => {
        const { getScripts, config } = this.props;
        const { project } = config;

        getScripts(project);
    }

    handleScriptChange = (name) => {
        const { getSingleScript, config } = this.props;
        const { project } = config;

        getSingleScript(project, name);
    }

    handleTextureChange = (name) => {
        const { onTextureChange = f => f, config } = this.props;
        const { project } = config;

        onTextureChange(project, name);
    }

    onPositionChange = (axis) => (value) => {
        const { onPositionChange, element, position, rotation, scale } = this.props;
        const newPosition = {
            ...position,
            [axis]: value
        };

        onPositionChange(element, newPosition, rotation, scale);
    }

    onRotationChange = (axis) => (value) => {
        const { onRotationChange, element, position, rotation, scale } = this.props;
        const newRotation = {
            ...rotation,
            [axis]: value
        };

        onRotationChange(element, position, newRotation, scale);
    }

    onScaleChange = (axis) => (value) => {
        const { onScaleChange, element, position, rotation, scale } = this.props;
        const newScale = {
            ...scale,
            [axis]: value
        };

        onScaleChange(element, position, rotation, newScale);
    }

    render() {
        const {
            empty,
            element,
            position,
            rotation,
            scale,
            type,
            scripts,
            assets
        } = this.props;

        return (
            <Col
                span={4}
                className='sidebar'>

                <div className="box">
                    <p className="title">
                        <Icon className="icon" type="bars" />
                        <span>Hierarchy</span>
                        <DeleteButton />
                        <CopyButton />
                        <AddButton />
                        <SearchButton />
                    </p>
                    <div className="content">
                        <Hierarchy />
                    </div>
                </div>
                <div className='box'>
                    <p className="title">
                        <Icon className="icon" type="search" />
                        <span>Inspector</span>
                    </p>
                    <div className="content">
                        <Inspector
                            onPositionChange={this.onPositionChange}
                            onRotationChange={this.onRotationChange}
                            onScaleChange={this.onScaleChange}
                            onScriptsMount={this.handleScriptMount}
                            onScriptChange={this.handleScriptChange}
                            onTextureChange={this.handleTextureChange}
                            scripts={scripts}
                            empty={empty}
                            type={type}
                            element={element}
                            position={position}
                            rotation={rotation}
                            scale={scale}
                            assets={assets}
                        />
                    </div>
                </div>
            </Col>
        );
    }
}


const mapStateToProps = (state) => {
    const { rightsidebar, scripts = {}, assets, config } = state;
    const { empty = true, element = '', type = '', position = {}, rotation = {}, scale= {} } = rightsidebar;

    return {
        assets,
        empty,
        element,
        type,
        position,
        rotation,
        scale,
        config,
        scripts
    }
}

const mapDispatchToProps = (dispatch) => ({
    onPositionChange: (element, position, rotation, scale) => dispatch(meshChanged(element, position, rotation, scale)),
    onRotationChange: (element, position, rotation, scale) => dispatch(meshChanged(element, position, rotation, scale)),
    onScaleChange: (element, position, rotation, scale) => dispatch(meshChanged(element, position, rotation, scale)),
    onTextureChange: (project, name) => dispatch(textureChanged(project, name)),

    getScripts: (project) => dispatch(getScripts(project)),
    getSingleScript: (project, name) => dispatch(getSingleScript(project, name))
})

export default connect(mapStateToProps, mapDispatchToProps)(RightSidebar);
