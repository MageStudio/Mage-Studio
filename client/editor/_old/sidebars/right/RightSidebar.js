import React from 'react';
import {
    connect
} from 'react-redux';
import { Col, Icon } from 'antd';

import CopyButton from '../../common/CopyButton';
import DeleteButton from '../../common/DeleteButton';
import AddButton from '../../common/AddButton';
import SearchButton from '../../common/SearchButton';

import Hierarchy from './Hierarchy';
import Inspector from './Inspector';

import {
    meshChanged,
    textureChanged,
    materialChanged
} from '../../actions/currentMesh';

import {
    getScripts,
    loadSingleScript
} from '../../actions/scripts';

export class RightSidebar extends React.Component {

    constructor(props) {
        super(props);
    }

    handleScriptMount = () => {
        const { getScripts, config } = this.props;
        const { project } = config;

        getScripts(project);
    }

    handleScriptChange = (name) => {
        const { getSingleScript, config } = this.props;
        const { project } = config;

        getSingleScript(project, name);
    }

    handleTextureChange = (name) => {
        const { onTextureChange = f => f, config } = this.props;
        const { project } = config;

        onTextureChange(project, name);
    }

    handleMaterialChange = (name) => {
        const { onMaterialChange = f => f } = this.props;

        onMaterialChange(name);
    }

    handleNameChange = (e) => {
        const { onNameChange, position, rotation, scale } = this.props;
        
        onNameChange(e.target.value, position, rotation, scale);
    }

    handlePositionChange = (axis) => (value) => {
        const { onPositionChange, name, position, rotation, scale } = this.props;
        const newPosition = {
            ...position,
            [axis]: value
        };

        onPositionChange(name, newPosition, rotation, scale);
    }

    handleRotationChange = (axis) => (value) => {
        const { onRotationChange, name, position, rotation, scale } = this.props;
        const newRotation = {
            ...rotation,
            [axis]: value
        };

        onRotationChange(name, position, newRotation, scale);
    }

    handleScaleChange = (axis) => (value) => {
        const { onScaleChange, name, position, rotation, scale } = this.props;
        const newScale = {
            ...scale,
            [axis]: value
        };

        onScaleChange(name, position, rotation, newScale);
    }

    render() {
        const {
            empty,
            name,
            position,
            rotation,
            scale,
            type,
            scripts,
            assets
        } = this.props;

        return (
            <Col
                span={4}
                className='sidebar'>

                <div className="box">
                    <p className="title">
                        <Icon className="icon" type="bars" />
                        <span>Hierarchy</span>
                        <DeleteButton />
                        <CopyButton />
                        <AddButton />
                        <SearchButton />
                    </p>
                    <div className="content">
                        <Hierarchy />
                    </div>
                </div>
                <div className='box'>
                    <p className="title">
                        <Icon className="icon" type="search" />
                        <span>Inspector</span>
                    </p>
                    <div className="content">
                        <Inspector
                            onPositionChange={this.handlePositionChange}
                            onRotationChange={this.handleRotationChange}
                            onScaleChange={this.handleScaleChange}
                            onScriptsMount={this.handleScriptMount}
                            onScriptChange={this.handleScriptChange}
                            onTextureChange={this.handleTextureChange}
                            onMaterialChange={this.handleMaterialChange}
                            onNameChange={this.handleNameChange}
                            scripts={scripts}
                            empty={empty}
                            type={type}
                            name={name}
                            position={position}
                            rotation={rotation}
                            scale={scale}
                            assets={assets}
                        />
                    </div>
                </div>
            </Col>
        );
    }
}


const mapStateToProps = (state) => {
    const { rightsidebar, scripts = {}, assets, config } = state;
    const { empty = true, name = '', type = '', position = {}, rotation = {}, scale = {} } = rightsidebar;

    return {
        assets,
        empty,
        name,
        type,
        position,
        rotation,
        scale,
        config,
        scripts
    }
}

const mapDispatchToProps = (dispatch) => ({
    onPositionChange: (name, position, rotation, scale) => dispatch(meshChanged(name, position, rotation, scale)),
    onRotationChange: (name, position, rotation, scale) => dispatch(meshChanged(name, position, rotation, scale)),
    onScaleChange: (name, position, rotation, scale) => dispatch(meshChanged(name, position, rotation, scale)),
    onNameChange: (name, position, rotation, scale) => dispatch(meshChanged(name, position, rotation, scale)),
    onTextureChange: (project, name) => dispatch(textureChanged(project, name)),
    onMaterialChange: (name) => dispatch(materialChanged(name)),
    getScripts: (project) => dispatch(getScripts(project)),
    getSingleScript: (project, name) => dispatch(loadSingleScript(project, name))
})

export default connect(mapStateToProps, mapDispatchToProps)(RightSidebar);
