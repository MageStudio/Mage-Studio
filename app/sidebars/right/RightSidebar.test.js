import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import { RightSidebar } from './RightSidebar';

const defaultProps = {
    config: {
        project: 'test'
    }
};

const getSidebar = (props) =>  shallow(<RightSidebar { ...defaultProps } { ...props }/>);

describe('RightSidebar', () => {

    it('should render fine', () => {
        const component = getSidebar();
        expect(component).toMatchSnapshot();
    });

    describe('events', () => {

        it('should call onPositionChange when the position changes', () => {
            const onPositionChange = sinon.spy();
            const component = getSidebar({ onPositionChange });

            component.instance().handlePositionChange('x')();

            expect(onPositionChange.called).toBe(true);
        });

        it('should call onRotationChange when the rotation changes', () => {
            const onRotationChange = sinon.spy();
            const component = getSidebar({ onRotationChange });

            component.instance().handleRotationChange('x')();

            expect(onRotationChange.called).toBe(true);
        });

        it('should call onScaleChange when the scale changes', () => {
            const onScaleChange = sinon.spy();
            const component = getSidebar({ onScaleChange });

            component.instance().handleScaleChange('x')();

            expect(onScaleChange.called).toBe(true);
        });

        it('should call onScriptMount when the scripts are on screen', () => {
            const getScripts = sinon.spy();
            const component = getSidebar({ getScripts });

            component.find('Inspector').simulate('scriptsMount');

            expect(getScripts.called).toBe(true);
        });

        it('should call onSCriptchange when the scripts are changing', () => {
            const getSingleScript = sinon.spy();
            const component = getSidebar({ getSingleScript });

            component.find('Inspector').simulate('scriptChange');

            expect(getSingleScript.called).toBe(true);
        });

        it('should call onTextureChange when the texture changed', () => {
            const onTextureChange = sinon.spy();
            const component = getSidebar({ onTextureChange });

            component.find('Inspector').simulate('textureChange');

            expect(onTextureChange.called).toBe(true);
        });

        it('should onMaterialChange when the material changes', () => {
            const onMaterialChange = sinon.spy();
            const component = getSidebar({ onMaterialChange });

            component.find('Inspector').simulate('materialChange');

            expect(onMaterialChange.called).toBe(true);
        });
    });
});