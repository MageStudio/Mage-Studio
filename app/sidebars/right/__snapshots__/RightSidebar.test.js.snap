// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightSidebar should render fine 1`] = `
<Col
  className="sidebar"
  span={4}
>
  <div
    className="box"
  >
    <p
      className="title"
    >
      <Icon
        className="icon"
        type="bars"
      />
      <span>
        Hierarchy
      </span>
      <DeleteButton />
      <CopyButton />
      <AddButton />
      <SearchButton />
    </p>
    <div
      className="content"
    >
      <SearchTree />
    </div>
  </div>
  <div
    className="box"
  >
    <p
      className="title"
    >
      <Icon
        className="icon"
        type="search"
      />
      <span>
        Inspector
      </span>
    </p>
    <div
      className="content"
    >
      <Inspector
        onMaterialChange={[Function]}
        onPositionChange={[Function]}
        onRotationChange={[Function]}
        onScaleChange={[Function]}
        onScriptChange={[Function]}
        onScriptsMount={[Function]}
        onTextureChange={[Function]}
      />
    </div>
  </div>
</Col>
`;
